pkg/vm/cli/cli.go

```diff
@@ -286,16 +286,40 @@ func handleRun(c *ishell.Context) {
 		}
 		v.LoadArgs(method, params)
 	}
-	v.Run()
+	runVMWithHandling(c, v)
 	changePrompt(c, v)
 }
 
+// runVMWithHandling runs VM with handling errors and additional state messages.
+func runVMWithHandling(c *ishell.Context, v *vm.VM) {
+	err := v.Run()
+	if err != nil {
+		c.Err(err)
+		return
+	}
+
+	var message string
+	switch {
+	case v.AtBreakpoint():
```

Comment:
And I think this check can be moved to be the last one. These statements are evaluated in order in which they appear in the code, so if for some reason we have several of them it makes sense to prioritize them. `HasFailed()` is a top-priority thing, then goes `HasHalted()` and only after that we can be concerned with breakpoints.